ConsentViewController/Classes/SourcePointClient/GDPRPrivacyManagerViewResponse.swift

```diff
@@ -13,32 +13,61 @@ struct GDPRVendor: Decodable {
     }
 
     enum Keys: String, CodingKey {
-        case policyUrl, vendorId, name
+        case policyUrl, legIntUrl, vendorId, name
         case iabId
-        case iabSpecialPurposes, iabFeatures, iabSpecialFeatures
+        case iabSpecialPurposes, iabSpecialPurposesObjs, iabFeatures, iabSpecialFeatures, iabDataCategories
         case description, cookieHeader
         case vendorType
         case consentCategories, legIntCategories, disclosureOnlyCategories
     }
 
     struct Category: Decodable {
-        let type: GDPRCategory.CategoryType
+        let type: GDPRCategory.CategoryType?
         let iabId: Int?
         let name: String
+        var retention: String?
+        
+        enum CodingKeys: CodingKey {
+            case type
+            case iabId
+            case name
+            case retention
+        }
+        
+        init(from decoder: Decoder) throws {
+            let container: KeyedDecodingContainer<GDPRVendor.Category.CodingKeys> = try decoder.container(keyedBy: GDPRVendor.Category.CodingKeys.self)
```

Comment:
You don't need to use the full explicit declaration here, you can short into:
```swift
let container = try decoder.container(keyedBy: CodingKeys.self)
```

Same for each attribute, you can do the following:
```swift
self.type = try container.decodeIfPresent(GDPRCategory.CategoryType.self, forKey:  .type)
```

And finally, you can remove the `self.` since you're not shadowing any attribute within that call.